sql/plan/merge_join.go

```diff
@@ -178,6 +223,24 @@ func (i *mergeJoinIter) Next(ctx *sql.Context) (sql.Row, error) {
 			case res == 0:
 				nextState = msSelect
 			}
+		case msRejectNull:
+			j := i.scopeLen + i.parentLen
+			for j < len(i.fullRow) {
+				if i.fullRow[j] == nil {
+					if j < i.scopeLen+i.parentLen+i.leftRowLen {
```

Comment:
(minor/nit) line 230 took me a minute to think through for the merge logic. Maybe worth a comment?